src/lib.rs

```diff
@@ -253,6 +253,55 @@ pub fn precise_time_s() -> f64 {
     return (precise_time_ns() as f64) / 1000000000.;
 }
 
+/// An opaque structure representing a moment in time.
+///
+/// The only operation that can be performed on a `PreciseTimestamp` is the
+/// calculation of the `Duration` of time that lies between them.
+///
+/// A `PreciseTimestamp` may be generated by the `precise_time()` function.
+///
+/// # Examples
+///
+/// Repeatedly call a function for 1 second:
+/// ```rust
+/// # fn do_some_work() {}
+/// let start = time::precise_time();
+///
+/// while start.to(time::precise_time()) < Duration::seconds(1) {
+///     do_some_work();
+/// }
+/// ```
+#[derive(Copy, Clone)]
+pub struct PreciseTimestamp(u64);
+
+impl PreciseTimestamp {
+    /// Returns a `Duration` representing the span of time from the value of
+    /// `self` to the value of `later`.
+    ///
+    /// # Notes
+    ///
+    /// If `later` represents a time before `self`, the result of this method
+    /// is unspecified.
+    ///
+    /// If `later` represents a time more than 293 years after `self`, the
+    /// result of this method is unspecified.
+    #[inline]
+    pub fn to(&self, later: PreciseTimestamp) -> Duration {
+        // NB: even if later is less than self due to overflow, this will work
+        // since the subtraction will underflow properly as well.
+        //
+        // We could deal with the overflow when casting to an i64, but all that
+        // gets us is the ability to handle intervals of up to 584 years, which
+        // seems not very useful :)
+        Duration::nanoseconds((later.0 - self.0) as i64)
+    }
+}
+
+/// Returns a `PreciseTimestamp` representing the current moment in time.
+pub fn precise_time() -> PreciseTimestamp {
```

Comment:
This construction seems a little odd for two reasons I think:

* The name `precise_time` and the returned name `PreciseTimespec` differ.
* This doesn't follow our constructor conventions of `Foo::new()`. It does align with `precise_time_ns`, but perhaps a more official struct should have a more conventional construction location?

In general this whole crate needs an overhaul though, so I'm certainly not opposed to incremental improvements! Curious what you think though.